src/main.rs

```diff
@@ -74,15 +73,18 @@ fn interp(prog: &[u8], map: Vec<usize>) {
         stdout: std::io::stdout(),
     };
 
-    // Locations are lazily instantiated later as loop-headers are found.
-    let mut locs: HashMap<usize, Location<_>> = HashMap::new();
-    let mut mtt = MTBuilder::new().hot_threshold(2).init();
+    // Locations are created for every program point, even if a loop cannot start there.
+    let mut locs = Vec::with_capacity(icx.prog.len());
+    locs.resize_with(prog.len(), || Location::new());
```

Comment:
I think `clippy` will suggest simplifying this to `locs.resize_with(prog.len(), Location::new)`.

Reply:
Fixed in 184ab2a